ibc-clients/ics07-tendermint/types/src/error.rs

```diff
@@ -5,51 +5,33 @@ use core::time::Duration;
 use displaydoc::Display;
 use ibc_core_client_types::error::ClientError;
 use ibc_core_commitment_types::error::CommitmentError;
-use ibc_core_host_types::error::IdentifierError;
+use ibc_core_host_types::error::{DecodingError, IdentifierError};
 use ibc_primitives::prelude::*;
 use ibc_primitives::TimestampError;
-use tendermint::{Error as TendermintError, Hash};
+use tendermint::Hash;
 use tendermint_light_client_verifier::errors::VerificationErrorDetail as LightClientErrorDetail;
 use tendermint_light_client_verifier::operations::VotingPowerTally;
 use tendermint_light_client_verifier::Verdict;
 
 /// The main error type for the Tendermint light client
 #[derive(Debug, Display)]
 pub enum TendermintClientError {
-    /// invalid identifier: `{0}`
-    InvalidIdentifier(IdentifierError),
+    /// decoding error: `{0}`
+    Decoding(DecodingError),
+    /// identifier error: `{0}`
+    Identifier(IdentifierError),
     /// invalid client state trust threshold: `{description}`
     InvalidTrustThreshold { description: String },
     /// invalid clock drift; must be greater than 0
     InvalidMaxClockDrift,
     /// invalid client proof specs: `{0}`
     InvalidProofSpec(CommitmentError),
-    /// invalid raw client state: `{description}`
-    InvalidRawClientState { description: String },
-    /// invalid raw header error: `{0}`
-    InvalidRawHeader(TendermintError),
     /// invalid raw misbehaviour: `{description}`
     InvalidRawMisbehaviour { description: String },
```

Comment:
This variant can be removed. It seems to be incorrectly used in the places where it appears. We should either choose another variant or define a new one for those cases.

Reply:
[a53a04e](https://github.com/cosmos/ibc-rs/pull/1325/commits/a53a04eaa8e8b6b3c98a6d2c6d75b0044d93d04e)